src/2dplane/ObstacleGrid.cpp

```diff
@@ -24,6 +25,25 @@ Vector2i ObstacleGrid::gridSquareForLocation(const Vector2f &loc) const {
                     loc.y() / height() * discretizedHeight());
 }
 
+float ObstacleGrid::nearestObstacleDist(const Vector2f &state, float maxDist) const {
+    //x and y are the indices of the cell that state is located in
+    float x = (state.x() / (_width / _discretizedWidth));
+    float y = (state.y() / (_height / _discretizedHeight));
+    //here we loop through the cells around (x,y) to find the minimum distance of the point to the nearest obstacle
+    for (int i = x - maxDist; i < x + maxDist && i >= 0 && i < discretizedWidth(); i++) {
+        for (int j = y - maxDist; j < y + maxDist && j >= 0 && j < discretizedHeight(); j++) {
+            bool obs = obstacleAt(i, j);
+            if (obs) {
+                float dist = sqrt((x-i)*(x-i)+(y-j)*(y-j));
```

Comment:
`dist` needs to be scaled from discrete coordinates back to continuous coordinates as well.